src/main/java/uk/gov/hmcts/reform/civil/service/StateGeneratorService.java

```diff
@@ -31,15 +35,22 @@ public CaseState getCaseStateForEndJudgeBusinessProcess(CaseData data) {
             decision = null;
         }
 
-        if (decision == MAKE_AN_ORDER && !isBlank(directionsText)) {
+        if (Boolean.TRUE.equals(hasDismissedCase)) {
```

Comment:
There is no need to compare value of `hasDismissedCase` to static `Boolean.True` value. `hasDismissedCase` method return boolean so we can use just this value: `if (hasDismissedCase)` ...